lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -352,6 +370,39 @@ private void storeHmrcInheritedIdentity(
         }
     }
 
+    private boolean isHmrcInheritedIdentityWithStrongerVotPresent(
+            SignedJWT incomingInheritedIdentity, String userId) throws CredentialParseException {
+        try {
+            var vcStoreItem =
+                    verifiableCredentialService.getVcStoreItem(HMRC_MIGRATION_CRI, userId);
+            SignedJWT existingInheritedIdentity = SignedJWT.parse(vcStoreItem.getCredential());
+
+            var existingInheritedIdentityVot =
+                    userIdentityService.getVot(existingInheritedIdentity);
+            var incomingInheritedIdentityVot =
+                    userIdentityService.getVot(incomingInheritedIdentity);
+
+            var indexOfExistingVot =
+                    HMRC_PROFILES_BY_STRENGTH.indexOf(existingInheritedIdentityVot);
+            var indexOfIncomingVot =
+                    HMRC_PROFILES_BY_STRENGTH.indexOf(incomingInheritedIdentityVot);
+
+            if (indexOfExistingVot == -1 || indexOfIncomingVot == -1) {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "At least one of the existing (%s) or incoming (%s) VoTs from hmrc aren't expected",
+                                existingInheritedIdentityVot, incomingInheritedIdentityVot));
+            }
+
+            return indexOfExistingVot - indexOfIncomingVot < 0;
+        } catch (ParseException | IllegalArgumentException e) {
+            throw new CredentialParseException(
+                    String.format(
+                            "Encountered a parsing error while attempting to parse or compare credentials: %s",
+                            e.getMessage()));
```

Comment:
Maybe add another signature to `CredentialParseException` so you can just pass the exception as a cause, rather than inlining the message?